app/helpers/search_helper.rb

```diff
@@ -14,4 +14,56 @@ def view_online(result)
 
     link_to 'View online', result['sourceLink'], class: 'button button-primary green'
   end
+
+  def view_record(record_id)
+    link_to 'View full record', record_path(id: record_id), class: 'button button-primary green'
+  end
+
+  def number_in_results(result)
+    result_index = 0
+    @results.each_with_index do |record, record_index|
+      result_index = record_index if record['timdexRecordId'] == result['timdexRecordId']
+    end
+    result_index + @pagination[:start]
+  end
+
+  # 'Coverage' and 'issued' seem to be the most prevalent types; 'coverage' is typically formatted as
+  # 'YYYY', whereas 'issued' comes in a variety of formats.
+  def parse_geo_dates(dates)
+    relevant_dates = if dates&.any? { |date| date['kind'] == 'Issued' }
+                       dates.select { |date| date['kind'] == 'Issued' }&.uniq
+                      elsif dates&.any? { |date| date['kind'] == 'Coverage' }
+                       dates.select { |date| date['kind'] == 'Coverage' }&.uniq
+                     end
+    return if relevant_dates.blank?
+
+    # Ignore dates of the same kind with unique values.
+    return if relevant_dates.length > 1
+
+    relevant_date = relevant_dates.first['value']
+
+    # If the date vaguely resembes a year, return it as is.
+    return relevant_date if relevant_date.length == 4
+
+    # If the date vaguely resembles 'YYYY-MM', 12/01/2020, or another unparsable date, extract the year.
+    handle_unparsable_date(relevant_date)
```

Comment:
I wonder if we can rely on some built-ins to coerce these strings into dates and then extract the years.

Reply:
Alternatively, I wonder if our GraphQL should have a field to return the `as_date` version of this field.

Reply:
`Date.strptime` provides some convenience, but you still have to infer the input format somehow in order to know how to parse it.